### PR TITLE
DB/Quest: Quest 5742 moved from core to DB

### PR DESCRIPTION
**Pull request by pangolp**
##### CHANGES PROPOSED:
- Add smart_scripts
- Remove c++ script

##### TESTS PERFORMED:
- Test on Windows.

##### HOW TO TEST THE CHANGES:
To test the changes, it is enough to take the mission.
.quest add 5742

##### KNOWN ISSUES AND TODO LIST:
None in particular, just change the c++ script to an SQL script

##### Target branch(es):
Master

### CODE REVIEW

**Comment by Winfidonarleyan** [comment]
Gossip is exist?

**Comment by pangolp** [comment]
@poszer The truth is I don't know how to put them together.  I usually armed them with event_type, action_type and target_type.  If that's how it works, he armed them.

@Winfidonarleyan Yes, the gossip exist in the db but could be added if necessary.  In fact, I have them in Spanish, so even the translations are in it.

I'll upload those modifications later then.  Maybe I was delayed by Monday or Tuesday at the latest.

**Comment by pangolp** [comment]
I'm going to review examples of smart_scripts comments that use the same event_type, action_type and target_type.  The truth is that sometimes in the rush like I forgot.  But basically when using gossip, the user closes the gossip and the quest is given completely.  The good thing is that it is translated.  Each contrary is seen in English.

**Comment by pangolp** [comment]
I have only one doubt, on the same file .sql I make the modifications and when doing the pull it relates to this PR or should I do another? The truth is that I have never had to edit or modify the code of a PR. Thank you.

File: data/sql/updates/pending_db_world/rev_1548473216184905600.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1548473216184905600');
+
+UPDATE `creature_template` SET `ScriptName`='', `AIName`='SmartAI' WHERE `entry`=1855;
+
+DELETE FROM `smart_scripts` WHERE `entryorguid`=1855 AND `source_type`=0;
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`event_param5`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_param4`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(1855,0,0,1,62,0,100,0,3683,0,0,0,0,72,0,0,0,0,0,0,7,0,0,0,0,0,0,0,0,""),
```
Comment (poszer) [inline data/sql/updates/pending_db_world/rev_1548473216184905600.sql]: I'll just give you **Example** : 
(1855,0,0,1,62,0,100,0,3683,0,0,0,0,72,0,0,0,0,0,0,7,0,0,0,0,0,0,0,0,"**Show Gossip Menu for 3683**"),

Fill ''comment'' row with text description

File: data/sql/updates/pending_db_world/rev_1548473216184905600.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+(1855,0,1,0,61,0,100,0,0,0,0,0,0,15,5742,0,0,0,0,0,7,0,0,0,0,0,0,0,0,"");
```
Comment (poszer) [inline data/sql/updates/pending_db_world/rev_1548473216184905600.sql]: I'll just give you **Example** : 
(1855,0,1,0,61,0,100,0,0,0,0,0,0,15,5742,0,0,0,0,0,7,0,0,0,0,0,0,0,0,"**Show Gossip Menu for 5742**");

Same here, Fill ''comment'' row with text description

**Comment by pangolp** [comment]
Okay, thank you. When I get home after work I will try to make the modifications and upload them. Anyway if you can go testing the script would be great. Although I should have also added the npc information, besides the quest.

**Comment by BarbzYHOOL** [comment]
To update your PR, you edit the files, you commit, and you push ;)

**Comment by pangolp** [comment]
I think I made it. I didn't do it before because I was working.

**Comment by pangolp** [comment]
Export the new table. Doesn't that parameter exist?

**Comment by poszer** [comment]
> Export the new table. Doesn't that parameter exist?
> [](http://subefotos.com/ver/?18ff36a65b24d978c3e4f42b87193955o.png)

Its okay, with updates param5 exist. Sorry, my bad

**Comment by poszer** [comment]
Tested, the SAI is okay. Gossip also exist. Thanks @pangolp

**Comment by FrancescoBorzi** [comment]
good job @pangolp and @poszer

**Comment by BarbzYHOOL** [comment]
@pangolp thank you!!!

@poszer thank you for testing!

**Pull request merged**